transip/client.py

```diff
@@ -122,17 +128,32 @@ def build_cookie(self, method, mode, parameters=None):
         timestamp = int(time.time())
         nonce = str(uuid.uuid4())[:32]
 
-        signature = self._sign(self._build_signature_message(
-            service_name=self.service_name, method_name=method,
-            timestamp=timestamp, nonce=nonce, additional=parameters))
+        signature = self._build_signature_message(
+            service_name=self.service_name,
+            method_name=method,
+            timestamp=timestamp,
+            nonce=nonce,
+            additional=parameters
+        )
+
+        signed_signature = self._sign(signature)
```

Comment:
If you really want to unwrap the signing here, I would propose different naming. `signed_signature` is kind of defying the point. This _is_ the signature.

```python
message_to_sign = self._build_signature_message(
    service_name=self.service_name, method_name=method,
    service_name=self.service_name,
    timestamp=timestamp, nonce=nonce, additional=parameters))
    method_name=method,
    timestamp=timestamp,
    nonce=nonce,
    additional=parameters
)
signature = self._sign(message_to_sign)
```

Something like that?

Reply:
Hehe, makes sense. Thanks for pointing out!

Reply:
I agree with mhogerheijde. Can you change 'signed_signature' to just 'signature'.